src/quicklist.c

```diff
@@ -1237,14 +1243,17 @@ quicklistIter *quicklistGetIterator(const quicklist *quicklist, int direction) {
  * return nodes in 'direction' direction. */
 quicklistIter *quicklistGetIteratorAtIdx(const quicklist *quicklist,
                                          const int direction,
-                                         const long long idx) {
+                                         const long long idx,
+                                         int safe)
+{
     quicklistEntry entry;
 
     if (quicklistIndex(quicklist, idx, &entry)) {
-        quicklistIter *base = quicklistGetIterator(quicklist, direction);
+        quicklistIter *base = quicklistGetIterator(quicklist, direction, safe);
         base->zi = NULL;
         base->current = entry.node;
         base->offset = entry.offset;
+        base->safe = safe;
```

Comment:
isn't the creation of the iterator already doing that (write to `iter->safe = safe`)?
```c
        quicklistIter *base = quicklistGetIterator(quicklist, direction, safe);
```

Reply:
ooh, indeed redundant.